subsys/bluetooth/host/hci_core.c

```diff
@@ -5481,44 +5547,81 @@ static uint8_t bt_read_static_addr(bt_addr_le_t *addr)
 		BT_WARN("Failed to read static addresses");
 		return 0;
 	}
+
+	if (IS_ENABLED(CONFIG_BT_HCI_VS_EXT_DETECT) &&
+	    rsp->len < sizeof(struct bt_hci_rp_vs_read_static_addrs)) {
+		BT_WARN("Invalid Vendor HCI extensions");
+		net_buf_unref(rsp);
+		return 0;
+	}
+
 	rp = (void *)rsp->data;
 	cnt = MIN(rp->num_addrs, CONFIG_BT_ID_MAX);
 
+	if (IS_ENABLED(CONFIG_BT_HCI_VS_EXT_DETECT) &&
+	    rsp->len != (sizeof(struct bt_hci_rp_vs_read_static_addrs) +
+			 cnt * sizeof(struct bt_hci_vs_static_addr))) {
```

Comment:
This isn't quite right if `CONFIG_BT_ID_MAX` is less than `rp->num_addrs`. So what you really should be multiplying by here is `rp->num_addrs` and not `cnt`

Reply:
Fixed.